fix: issue on dependencies auro--select for matchWidth for the value to much long

irma-kurnia-alaska opened the pull request:
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** [Demo for flexible width containers #101](https://github.com/AlaskaAirlines/auro-select/issues/101)

## Summary:

- Set the triggerContentWrapper css to have truncated ui
- set  label css have white-space: nowrap (handle when the label have more text)

I attached the screenshot in my local base the modified:
<img width="270" alt="sfdfd" src="https://github.com/AlaskaAirlines/auro-dropdown/assets/124669886/d3b11e1f-4f42-42fd-90f6-6a1f691d7e89">

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)


## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

irma-kurnia-alaska committed:
fix: issue on dependencies auro-select for matchWidth for the value text to much long

Change to be committed:
modified: package.json
modified: src/style.scss

fajar-apri-alaska commented on package.json:
```diff
@@ -7,7 +7,7 @@
   ],
   "name": "@aurodesignsystem/auro-dropdown",
-  "version": "2.10.1",
+  "version": "2.10.2",
```
no need to update this manually because Auro use semantic versioning.


edit:
oops, I mean semantic release.

irma-kurnia-alaska replied:
oh, okay, thanks @fajar-apri-alaska, i will revert this.

blackfalcon reviewed (changes requested):
Please add an example of this feature in the demo or apiExamples.

irma-kurnia-alaska committed:
docs: add new demo example for truncated text

Change to be committed:
new fille: apiExamples/truncatedText.html
modified: demo/demo.md
modified: docs/partials/demo.md

fajar-apri-alaska committed:
perf: update scripts

- update to __dirname declaration to also work on MacOS machines.

Changes to be committed:
	modified:   scripts/generateDocs.mjs

fajar-apri-alaska committed:
docs: update to truncatedText example

Changes to be committed:
	modified:   apiExamples/truncatedText.html
	modified:   demo/demo.md
	modified:   docs/partials/demo.md

jason-capsule42 commented on scripts/generateDocs.mjs:
```diff
@@ -2,8 +2,9 @@ import path from 'path';
 import fs from 'fs';
 import https from 'https';
+import { fileURLToPath } from 'url';
 
-const __dirname = new URL('.', import.meta.url).pathname;
+const __dirname = fileURLToPath(new URL('.', import.meta.url));
```
This file is coming from a template that all components use and individual components should not diverge.

Is there a case why this change is necessary for this issue?

fajar-apri-alaska replied:
@jason-capsule42, in my case, it doesn't work on my machine.
I have something like `home/fajar/Git Project/auro-dropdown` local path. When I ran the old script, it returns `home/fajar/Git%20Project/auro-dropdown` and somehow it couldn't find the path it refer to.

<img width="1129" alt="image" src="https://github.com/AlaskaAirlines/auro-dropdown/assets/124668226/501ce4f2-af21-4cf8-82b3-6161ebbbb383">

I already asked Dale about this as well, He said it is fine to change that.

Reference: https://github.com/AlaskaAirlines/auro-radio/pull/120/commits/2fcaa20682b749e8bb7fb9340e7f1492db3c19b9

jason-capsule42 replied:
Let't not make this change as a one off just to this component repo. @jordanjones243 @blackfalcon what do you think about this proposed change to the script. Is this something you think makes sense to change to all the generateDocs.js files in all components?

jason-capsule42 commented on docs/partials/index.md:
```diff
@@ -106,6 +106,22 @@ This common example uses the default `auro-dropdown` element with the attributes
 
 </auro-accordion>
 
+#### Truncated trigger component with fixed width
```
Let's move this to docs/partials/api.md and put it with the "Other Examples" section of the page.

It does not belong here as a subjection of this heading.

irma-kurnia-alaska replied:
ok @jason-capsule42, thanks.

jason-capsule42 reviewed (changes requested):
See comments and also rebase the branch.

irma-kurnia-alaska committed:
docs: update feedback to move new docs into docs//partial/api.md

Change to be committed:
modified: demo/api.md
modified: demo/index.md
modified: docs/partials/api.md
modified: docs/partials/index.md

blackfalcon commented:
:tada: This PR is included in version 2.10.6 :tada:

The release is available on:
- [npm package (@latest dist-tag)](https://www.npmjs.com/package/@aurodesignsystem/auro-dropdown/v/2.10.6)
- [GitHub release](https://github.com/AlaskaAirlines/auro-dropdown/releases/tag/v2.10.6)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
